### PR TITLE
Add maven readme

### PR DESCRIPTION
**Pull request by tanzoniteblack**

### CODE REVIEW

File: build_tools/maven.md [hunk 1]
```diff
@@ -5,4 +5,44 @@ order: 4
...
+  </pluginGroups>
+```
+
+## Generate `.ensime` file
+
+### (Optional) Download project sources
+
+The ensime-maven plugin will tell ensime about the location of source jars, but won't automatically download them for you. You can get maven to do this by running:
+
+``` shell
```
Comment (fommil) [inline build_tools/maven.md]: typo

Comment (tanzoniteblack) [inline build_tools/maven.md]: You're going to have to be more explicit about what the typo is, I'm not seeing it.

Comment (fommil) [inline build_tools/maven.md]: I thought it was the space, but I think `shell` is not supported.

Comment (fommil) [inline build_tools/maven.md]: (I fixed it already :smile:) thanks for porting this over!

**Pull request merged**